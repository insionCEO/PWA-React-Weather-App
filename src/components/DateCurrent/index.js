import React, { Component } from 'react';
import rAFTimeout from '../../helpers/rAFTimeout';
import './index.scss';

class DateCurrent extends Component {
  constructor() {
    super();

    this.text = React.createRef();
  }

  animate() {
    rAFTimeout(() => this.text.current.classList.add('animate-in'), 400);
  }

  componentDidMount() {
    rAFTimeout(() => this.animate(), 350);
  }

  render() {
    return (
      <section className="date-current">
        <span ref={this.text} className="date-current__text">{this.props.date}</span>
      </section>
    )
  }
}

export default DateCurrent;
