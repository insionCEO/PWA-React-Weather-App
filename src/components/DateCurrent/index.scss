.date-current {
  align-items: center;
  color: rgba(255, 255, 255, .8);
  display: flex;
  font-size: 18px;
  font-size: 16px;
  justify-content: center;
  left: 50%;
  position: absolute;
  transform: translate(-50%, 80px);

  &__text {
    opacity: 0;
    transform: translateY(-50px);
    transition: transform .3s cubic-bezier(.165, .84, .44, 1), opacity .3s .1s ease-out;
    will-change: opacity, transform;
  }

  &__text.animate-in {
    opacity: 1;
    transform: translateY(0);
  }
}
