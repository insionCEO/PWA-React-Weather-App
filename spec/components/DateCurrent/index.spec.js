import React from 'react';
import DateCurrent from '../../../src/components/DateCurrent';
import { enzymeConfig, mount } from '../../enzymeConfig';

enzymeConfig();

describe('DateCurrent', () => {
  it('matches snapshot', () => {
    const component = mount(<DateCurrent />);

    expect(component).toMatchSnapshot();
  });

  describe('componentDidMount', () => {
    it('calls animate', (done) => {
      const component = mount(<DateCurrent />);
      const instance = component.instance();

      instance.animate = jest.fn();

      setTimeout(() => {
        expect(instance.animate).toBeCalled();
        done();
      }, 400);
    });
  })
});
