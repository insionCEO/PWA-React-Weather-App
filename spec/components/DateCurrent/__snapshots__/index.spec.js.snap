// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateCurrent matches snapshot 1`] = `
<DateCurrent>
  <section
    className="date-current"
  >
    <span
      className="date-current__text"
    />
  </section>
</DateCurrent>
`;
